### PR TITLE
feat: more explicit display of XSLT params

### LINKED ISSUE #92: Improve parameter display
In the select step, XSLT parameters are displayed like other fields, but should be sub-fields of the XSLT drop-down.

### PR DESCRIPTION
**Pull request by streino**
Fix #92 

image

### CODE REVIEW

File: isomorphe/templates/fragments/select_transformation.html.j2 [hunk 1]
```diff
@@ -1,9 +1,16 @@
 {% set params = transformation.params %}
 {% if params %}
+<div>
+Paramètres
+<ul>
 {% for param in params %}
+<li>
   <div class="fr-input-group">
-    <label class="fr-label" for="query">{{ param.name|capitalize }}{% if param.required %} *{% endif %}</label>
+    <label class="fr-label" for="query">{{ param.name }} : {% if param.required %} *{% endif %}</label>
```
Comment (streino) [inline isomorphe/templates/fragments/select_transformation.html.j2:9]: Removed `|capitalize` so it matches whatever is in the xslt.

**Review by abulte — COMMENTED** [review]
Did you try a `<fieldset>`? I can't find an example on https://www.systeme-de-design.gouv.fr/composants-et-modeles/composants/champ-de-saisie but it's mentioned, and `vue-dsfr` has one https://vue-ds.fr/composants/DsfrFieldset for inspiration.

**Comment by streino** [comment]
> Did you try a `<fieldset>`? I can't find an example on https://www.systeme-de-design.gouv.fr/composants-et-modeles/composants/champ-de-saisie but it's mentioned, and `vue-dsfr` has one https://vue-ds.fr/composants/DsfrFieldset for inspiration.

Different attempts :

```
<DsfrFieldset>
<legend>Paramètres</legend>
{% for param in params %}
  <div class="fr-input-group">
    <label class="fr-label" for="query">{{ param.name }} : {% if param.required %} *{% endif %}</label>
    <input {% if param.required %}required{% endif %} class="fr-input" type="text" id="param-{{ param.name }}" name="param-{{ param.name }}" value="{{ param.default_value }}" />
  </div>
{% endfor %}
</DsfrFieldset>
```

image

```
<DsfrFieldset>
<legend>Paramètres</legend>
{% for param in params %}
    <label class="fr-label" for="query">{{ param.name }} : {% if param.required %} *{% endif %}</label>
    <input {% if param.required %}required{% endif %} class="fr-input" type="text" id="param-{{ param.name }}" name="param-{{ param.name }}" value="{{ param.default_value }}" />
{% endfor %}
</DsfrFieldset>
```

image

```
<fieldset class="fr-fieldset" id="display-fieldset">
<legend class="fr-fieldset__legend--regular fr-fieldset__legend" id="display-fieldset-legend">
Paramètres
</legend>
{% for param in params %}
<div class="fr-fieldset__element">
    <label class="fr-label" for="query">{{ param.name }} : {% if param.required %} *{% endif %}</label>
    <input {% if param.required %}required{% endif %} class="fr-input" type="text" id="param-{{ param.name }}" name="param-{{ param.name }}" value="{{ param.default_value }}" />
</div>
{% endfor %}
</fieldset>
```
(grep'ed "fieldset" in https://www.systeme-de-design.gouv.fr/composants-et-modeles/composants/parametre-d-affichage/)

image

None look too convincing, but I have no idea what I'm doing :keyboard: :monkey: Any better idea?

**Comment by abulte** [comment]
The default one is not too bad.

```html
<fieldset>
  <legend><strong>Paramètres</strong></legend>
  [...]
</fieldset>
```

Capture d’écran 2024-11-03 à 10 11 45

The border style is a bit strange. You can play with it:

```css
border-width: 1px;
border-style: solid;
```

**Comment by abulte** [comment]
@streino did it in https://github.com/ecolabdata/ecospheres-isomorphe/pull/93/commits/832480ae851be05c21d10c685f7dc4bbcf354dd4

**Review by abulte — APPROVED** [review]

**Comment by streino** [comment]
> @streino did it in [832480a](https://github.com/ecolabdata/ecospheres-isomorphe/commit/832480ae851be05c21d10c685f7dc4bbcf354dd4)

Thanks! Looks good :)

**Pull request merged**